src/GafferOSL/ShadingEngine.cpp

```diff
@@ -294,6 +294,22 @@ class RenderState
 				t.arraylen = 2;
 				return ShadingSystem::convert_value( value, type, src, t );
 			}
+			else if ( it->dataView.type.basetype == TypeDesc::DOUBLE )
```

Comment:
This would accept a V3d as well as just a double, right? Is that intended? I don't think we support a V3f->float conversion because it's ambiguous, and I would expect the same to apply here.

Reply:
checked we're dealing with a scalar double in ba245a1